### PR TITLE
[project-test] Use temple test as an integration test

### PR DESCRIPTION
**Pull request by jaylees14**
I gift `temple test` as an integration test.

Very simple usage: just check that no exceptions were thrown.
Probably should only use with `docker-compose` for speed, not sure how well `Kubernetes` will work on Travis either.

Also:
- Found a typo in some of the min/max logic, so I fixed that.
- Added a consistent timeout to each integration test

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
LGTM

File: src/it/scala/temple/containers/EndpointTesterSpec.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import temple.DSL.semantics.Analyzer
+import temple.builder.project.ProjectBuilder
+import temple.detail.LanguageSpecificDetailBuilder
+import temple.test.ProjectTester
+import temple.utils.FileUtils
+import temple.utils.MonadUtils.FromEither
+
+abstract class EndpointTesterSpec extends FlatSpec {
+
+  def testEndpoints(templefile: String): Unit = {
```
Comment (c-harding) [inline src/it/scala/temple/containers/EndpointTesterSpec.scala]: `buildAndTestEndpoints`? Reading the test, it sounds like it tests something that’s already been set up

File: src/it/scala/temple/containers/HadolintSpec.scala [hunk 2]
```diff
@@ -19,6 +19,10 @@ abstract class HadolintSpec extends DockerShell2HttpService(8080) with DockerTes
   // Validate a given Dockerfile, returning the output of Hadolint
   def validate(dockerfile: String): String = {
     val json = Map("contents" -> dockerfile).asJson.toString()
-    Http(hadolintVerifyUrl).params(Map("dockerfile" -> json)).asString.body
+    Http(hadolintVerifyUrl)
+      .params(Map("dockerfile" -> json))
+      .timeout(connTimeoutMs = 1000, readTimeoutMs = 30000)
+      .asString
+      .body
```
Comment (c-harding) [inline src/it/scala/temple/containers/HadolintSpec.scala:26]: This pattern is used so often, why not add it as a method in DockerShell2HttpService?

File: src/main/scala/temple/ast/AttributeType.scala [hunk 3]
```diff
@@ -42,7 +42,7 @@ object AttributeType {
       else Float.MaxValue
     }
 
-    def minValue: Double = max.getOrElse {
+    def minValue: Double = min.getOrElse {
```
Comment (c-harding) [inline src/main/scala/temple/ast/AttributeType.scala:45]: My bad, thank you

**Review by c-harding — APPROVED** [review]
nice

File: src/it/scala/temple/containers/EndpointTesterSpec.scala (hunk 1 above)
Comment (jaylees14) [reply src/it/scala/temple/containers/EndpointTesterSpec.scala]: Good idea

File: src/it/scala/temple/containers/HadolintSpec.scala (hunk 2 above)
Comment (jaylees14) [reply src/it/scala/temple/containers/HadolintSpec.scala:26]: Shall do in a follow up

**Pull request merged**